### PR TITLE
USB 3.0 device side support

### PR DESCRIPTION
**Pull request by yangsong8-ai**

### CODE REVIEW

**Review by acassis — APPROVED** [review]

**Comment by acassis** [comment]
@yangsong8-ai is there some MCU or processor with USB 3.0 supported by NuttX ?

**Comment by xiaoxiang781216** [comment]
> @yangsong8-ai is there some MCU or processor with USB 3.0 supported by NuttX ?

it's a new device does not release yet.

File: drivers/usbdev/cdcacm.c [hunk 1]
```diff
@@ -1021,18 +1001,8 @@ static int cdcacm_setconfig(FAR struct cdcacm_dev_s *priv, uint8_t config)
...
-                               &priv->devinfo, true);
-    }
-  else
-#endif
-    {
-      ret = cdcacm_epconfigure(priv->epbulkout, CDCACM_EPBULKOUT, true,
-                               &priv->devinfo, false);
-    }
+  ret = cdcacm_epconfigure(priv->epbulkout, CDCACM_EPBULKOUT, true,
+                            &priv->devinfo, priv->usbdev->speed);
```
Comment (xiaoxiang781216) [inline drivers/usbdev/cdcacm.c:1005]: align

File: drivers/usbdev/cdcacm.c [hunk 2]
```diff
@@ -998,18 +988,8 @@ static int cdcacm_setconfig(FAR struct cdcacm_dev_s *priv, uint8_t config)
...
-                               &priv->devinfo, true);
-    }
-  else
-#endif
-    {
-      ret = cdcacm_epconfigure(priv->epbulkin, CDCACM_EPBULKIN, false,
-                               &priv->devinfo, false);
-    }
+  ret = cdcacm_epconfigure(priv->epbulkin, CDCACM_EPBULKIN, false,
+                            &priv->devinfo, priv->usbdev->speed);
```
Comment (xiaoxiang781216) [inline drivers/usbdev/cdcacm.c:992]: ditto

File: drivers/usbdev/cdcacm.c [hunk 3]
```diff
@@ -975,18 +975,8 @@ static int cdcacm_setconfig(FAR struct cdcacm_dev_s *priv, uint8_t config)
...
-                               &priv->devinfo, true);
-    }
-  else
-#endif
-    {
-      ret = cdcacm_epconfigure(priv->epintin, CDCACM_EPINTIN, false,
-                               &priv->devinfo, false);
-    }
+  ret = cdcacm_epconfigure(priv->epintin, CDCACM_EPINTIN, false,
+                            &priv->devinfo, priv->usbdev->speed);
```
Comment (xiaoxiang781216) [inline drivers/usbdev/cdcacm.c:979]: ditto

**Comment by masayuki2009** [comment]
@yangsong8-ai 
Hmm, I noticed that spresense:rndis does not work now with this PR.

```
nsh> ps
  PID GROUP PRI POLICY   TYPE    NPX STATE    EVENT     SIGMASK           STACK   USED  FILLED COMMAND
    0     0   0 FIFO     Kthread   - Ready              0000000000000000 001008 000512  50.7%  Idle_Task
    1     0 224 RR       Kthread   - Waiting  Semaphore 0000000000000000 001984 000312  15.7%  hpwork 0x2d061568 0x2d06157c
    2     0 100 RR       Kthread   - Waiting  Semaphore 0000000000000000 001984 000312  15.7%  lpwork 0x2d061540 0x2d061554
    3     3 100 RR       Task      - Running            0000000000000000 003032 001240  40.8%  spresense_main
    4     4 200 RR       Task      - Waiting  MQ empty  0000000000000000 000984 000464  47.1%  cxd56_pm_task
    6     6 100 RR       Task      - Waiting  Semaphore 0000000000000000 002016 000632  31.3%  telnetd
nsh> free
                 total       used       free    maxused    maxfree  nused  nfree
      Umem:    1131436      42524    1088912      43856    1087952    105      2
nsh> ifconfig
lo	Link encap:Local Loopback at RUNNING mtu 1518
	inet addr:127.0.0.1 DRaddr:127.0.0.1 Mask:255.0.0.0

eth0	Link encap:Ethernet HWaddr 00:e0:de:ad:be:ff at UP mtu 1500
	inet addr:10.0.0.2 DRaddr:10.0.0.1 Mask:255.255.255.0

             IPv4   TCP   UDP  ICMP
Received     0000  0000  0000  0000
Dropped      0000  0000  0000  0000
  IPv4        VHL: 0000   Frg: 0000
  Checksum   0000  0000  0000  ----
  TCP         ACK: 0000   SYN: 0000
              RST: 0000  0000
  Type       0000  ----  ----  0000
Sent         0000  0000  0000  0000
  Rexmit     ----  0000  ----  ----
nsh> ifconfig eth0 10.0.1.20
nsh> ifconfig
lo	Link encap:Local Loopback at RUNNING mtu 1518
	inet addr:127.0.0.1 DRaddr:127.0.0.1 Mask:255.0.0.0

eth0	Link encap:Ethernet HWaddr 00:e0:de:ad:be:ff at UP mtu 1500
	inet addr:10.0.1.20 DRaddr:10.0.1.1 Mask:255.255.255.0

             IPv4   TCP   UDP  ICMP
Received     0000  0000  0000  0000
Dropped      0000  0000  0000  0000
  IPv4        VHL: 0000   Frg: 0000
  Checksum   0000  0000  0000  ----
  TCP         ACK: 0000   SYN: 0000
              RST: 0000  0000
  Type       0000  ----  ----  0000
Sent         0000  0000  0000  0000
  Rexmit     ----  0000  ----  ----
nsh> ping -c 10 10.0.1.15
PING 10.0.1.15 56 bytes of data
ERROR: sendto failed at seqno 0: 101
ERROR: sendto failed at seqno 1: 101
ERROR: sendto failed at seqno 2: 101
ERROR: sendto failed at seqno 3: 101
ERROR: sendto failed at seqno 4: 101
ERROR: sendto failed at seqno 5: 101
ERROR: sendto failed at seqno 6: 101
ERROR: sendto failed at seqno 7: 101
ERROR: sendto failed at seqno 8: 101
ERROR: sendto failed at seqno 9: 101
```

**Comment by xiaoxiang781216** [comment]
@yangsong8-ai please take a look.

**Comment by yangsong8-ai** [comment]
OK，I am checking what's the problem.

**Comment by masayuki2009** [comment]
@yangsong8-ai 
I also noticed that stm32f4discovery:rndis does not work.
Shall we revert this PR?

**Comment by masayuki2009** [comment]
> Shall we revert this PR?

Hmm, we can not revert this PR...

**Comment by yangsong8-ai** [comment]
@masayuki2009    I have found a spresense board and am doing environment building to analyze this.

**Comment by yangsong8-ai** [comment]
@masayuki2009  I have updataed patch https://github.com/apache/nuttx/pull/13251 for this issue.  Thanks for your help.

**Pull request merged**